### ISSUE TITLE
CLI: prisma generate fails as post-deploy hook

### ISSUE THREAD

**Issue by RobertShaw1** [prisma/prisma1#4156]
**Describe the bug**
When installing prisma via homebrew the post-deploy hook `prisma generate` fails because of a missing module.

**To Reproduce**
Follow the [Get Started](https://www.prisma.io/docs/get-started/01-setting-up-prisma-new-database-JAVASCRIPT-a002/) tutorial 
1. Set up  Prisma
2. Change Datamodel
    * [Step 4 - Regenerate your Prisma Client](https://www.prisma.io/docs/get-started/02-change-data-model-JAVASCRIPT-c001/#regenerate-your-prisma-client)

After adding `prisma generate` as a post-deploy hook here is what my `prisma.yml` looks like...
![Screen Shot 2019-03-11 at 1 28 20 AM](https://user-images.githubusercontent.com/26239428/54104210-fec82700-439c-11e9-815a-369336b80e6e.png)

Below is the output upon executing `prisma deploy`...
![Screen Shot 2019-03-11 at 1 25 06 AM](https://user-images.githubusercontent.com/26239428/54104077-9bd69000-439c-11e9-94ef-4e52c45f90f9.png)

**Expected behavior**
When executing `prisma generate` on the command line, prisma successfully generates both the schema and the client in the `generated` directory.  I would expect the same behavior when `prisma generate` is executed as a `post-deploy` hook.

**Versions:**
 - Connector: `Postgres`
 - Prisma Server: `1.27.0`
 - `prisma` CLI: `prisma/1.28.0 (darwin-x64) node-v10.4.1`
 - OS:  `OS X Mojave 10.14.3`
 - other dependencies: `prisma-client`

**Comment by divyenduz** [comment]
@RobertShaw1 : Did you install Prisma using brew or npm?

**Comment by RobertShaw1** [comment]
I installed Prisma using brew.  This is mentioned at the very beginning of the issue.

**Comment by RobertShaw1** [comment]
@timsuchanek : What additional info is needed for this issue?

**Comment by divyenduz** [comment]
@RobertShaw1 : Thanks, this information is enough, I am able to reproduce. However, this would need more debugging from my end for a fix.

**Comment by RobertShaw1** [comment]
@divyenduz : Thanks.  I just wanted to make sure the issue didn't get closed or marked as stale because there was additional info I needed to submit.  Your efforts are much appreciated.

**Comment by pantharshit00** [comment]
@RobertShaw1 

Can you please try the new version(1.30) from homebrew?

**Comment by RobertShaw1** [comment]
@pantharshit00 
I got the same error message as described above.
![Screen Shot 2019-04-10 at 10 01 16 PM](https://user-images.githubusercontent.com/26239428/55927867-37496380-5bdc-11e9-9ea3-235f11f9b535.png)

**Comment by tricoder42** [comment]
I confirm this is happening in `1.31.0` (version `1.31.1` wasn't available in brew when I checked for updates).

As a workaround, appending `sh -c` solves the issue:

```yml
endpoint: http://localhost:4466
datamodel: datamodel.prisma

hooks:
  post-deploy:
    - sh -c prisma generate
```

I checked output of `which prisma` from post-deploy hook and it's the same as running it outside: `/usr/local/bin/prisma`. I also tried to remove `node_modules`/`package.json` just to confirm, that the app doesn't load any local files, but the error persists.
